### PR TITLE
Fix `CreateDefaultBuilder_SecretsDoesReload` test failure

### LINKED ISSUE #48696: CreateDefaultBuilder_SecretsDoesReload test failed in CI
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-48684-merge-1dda5503a115420687/Microsoft.Extensions.Hosting.Unit.Tests/console.b5b90d65.log?sv=2019-07-07&se=2021-03-16T05%3A35%3A14Z&sr=c&sp=rl&sig=ytXzgYX%2F6x0X9rBYl8sm%2B1JpNoEfEPjQFQhlMtH69tU%3D
```
  Discovering: Microsoft.Extensions.Hosting.Unit.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  Microsoft.Extensions.Hosting.Unit.Tests (found 99 test cases)
  Starting:    Microsoft.Extensions.Hosting.Unit.Tests (parallel test collections = on, max threads = 2)
[40m[32minfo[39m[22m[49m: Microsoft.Extensions.Hosting.Tests.HostTests[0]
      Request starting
    Microsoft.Extensions.Hosting.Tests.HostTests.CreateDefaultBuilder_SecretsDoesReload [FAIL]
      Assert.Equal() Failure
                                    â†“ (pos 20)
      Expected: Hello ASP.NET Core: f577b280d3814d6c935970a1aadbb8e6
      Actual:   Hello ASP.NET Core: 407a83ef08c04e1ca1e07690cb30dc0a
                                    â†‘ (pos 20)
      Stack Trace:
        /_/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs(264,0): at Microsoft.Extensions.Hosting.Tests.HostTests.CreateDefaultBuilder_SecretsDoesReload()
        --- End of stack trace from previous location ---
  Finished:    Microsoft.Extensions.Hosting.Unit.Tests
```

### PR DESCRIPTION
**Pull request by IEvangelist**
Test failures appear to be isolated to Ubuntu 16.04, I'm wondering if this is related to the `FileSystemWatcher` not signaling file system changes correctly.

- Sets the `"DOTNET_USE_POLLING_FILE_WATCHER"` env var to `"1"` for the duration of the test

Fixes #48696

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs [hunk 1]
```diff
@@ -275,44 +275,53 @@ static string SaveRandomConfig(string appSettingsPath)
...
         {
-            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
-            var reloadFlagConfig = new Dictionary<string, string>() { { "hostbuilder:reloadConfigOnChange", "true" } };
-            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
-            var secretFileInfo = new FileInfo(secretPath);
-
-            Directory.CreateDirectory(secretFileInfo.Directory.FullName);
-
-            static string SaveRandomSecret(string secretPath)
+            var originalValue = Environment.GetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER");
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs:278]: Note for review: The diff looks tough to read but seems like this change is basically just adding a try/finally block around this entire code snippet.

**Comment by ericstj** [comment]
If we rely on polling to be set doesn't that mean we have a problem with the product, since that is not the default?

File: src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs [hunk 2]
```diff
@@ -275,44 +275,53 @@ static string SaveRandomConfig(string appSettingsPath)
...
+            try
             {
-                var newMessage = $"Hello ASP.NET Core: {Guid.NewGuid():N}";
-                File.WriteAllText(secretPath, $"{{ \"Hello\": \"{newMessage}\" }}");
-                return newMessage;
-            }
+                Environment.SetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER", "1");
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs:281]: Do these tests run concurrently?  If so, then setting the environment variable for the entire process might effect other tests, you can use RemoteExecutor to give the test it's own process, or use API to setup the polling file-watcher.
https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/PhysicalFileProviderTests.cs#L117-L118

**Comment by IEvangelist** [comment]
> If we rely on polling to be set doesn't that mean we have a problem with the product, since that is not the default?

Hi @ericstj - yes, but I think this is a somewhat known issue for certain environments. In fact, I just had to update docs to mention details about this per @davidfowl, see https://github.com/dotnet/runtime/issues/51866#issuecomment-826936347. 

I'm not certain that this is isolated to Ubuntu 16.04, but that is the only environment that I saw occasionally failing for this test unit test. I believed it to be related to the `FileSystemWatcher` not reliably signaling that the file has changed. I also found an old [SO](https://stackoverflow.com/a/31236514/2410379) that mentioned concerns with Ubuntu and the `FileSystemWatcher` not signaling. I could be wrong, but I did want to at least give it a try.

**Comment by IEvangelist** [comment]
I'm going to abandon/close this PR. I was unable to reliably validate these changes, per https://github.com/dotnet/runtime/pull/52466#issuecomment-836655855.

**Pull request closed without merging**